src/features/layers_in_area/atoms/layersInAreaAndEventLayerResource.ts

```diff
@@ -0,0 +1,75 @@
+import { apiClient } from '~core/apiClientInstance';
+import {
+  currentApplicationAtom,
+  currentEventFeedAtom,
+  focusedGeometryAtom,
+} from '~core/shared_state';
+import { createAtom } from '~utils/atoms';
+import { createAsyncAtom } from '~utils/atoms/createAsyncAtom';
+import { removeEmpty } from '~utils/common';
+import { LAYERS_IN_AREA_API_ERROR } from '../constants';
+import type { FocusedGeometry } from '~core/shared_state/focusedGeometry';
+import type { LayerInArea } from '../types';
+
+type LayersInAreaAndEventLayerResourceParameters = {
+  appId: string;
+  geoJSON: GeoJSON.GeoJSON;
+  eventId?: string;
+  eventFeed?: string;
+};
+
+const getEventId = (focusedGeometry: FocusedGeometry) => {
+  return 'meta' in focusedGeometry.source && 'eventId' in focusedGeometry.source.meta
+    ? focusedGeometry.source.meta.eventId
+    : null;
+};
+
+const layersInAreaAndEventLayerResourceParametersAtom = createAtom(
+  {
+    currentApplicationAtom,
+    focusedGeometryAtom,
+  },
+  ({ get, getUnlistedState }): LayersInAreaAndEventLayerResourceParameters | null => {
+    const focusedGeometry = get('focusedGeometryAtom');
+    const appId = get('currentApplicationAtom');
+
+    // Check required
+    if (focusedGeometry === null) return null;
+    if (appId === null) return null;
+
+    const geoJSON = focusedGeometry.geometry;
+    const eventId = getEventId(focusedGeometry);
+    // getUnlistedState for avoid extra request when layer if
+    // Updates flow:
+    // 1. eventId changed (ignoring)
+    // 2. focusedGeometry start loading for new event
+    // 3. focusedGeometry loaded
+    // 4. get layers for new geometry
+    const eventFeed = getUnlistedState(currentEventFeedAtom)?.id;
+
+    return removeEmpty({
+      appId,
+      eventFeed,
+      eventId,
+      geoJSON,
+    });
+  },
+);
+
+export const layersInAreaAndEventLayerResource = createAsyncAtom(
```

Comment:
Why do we need to add that EventLayer is here too? considering we're getting the layers related to event-id and event-feed-id?

Reply:
because it true. 
We get 1) layers in area and 2) event layer here (in ui in named as "event shape". In near feature we will get event shape from other api in other feature